File: playwright_framework/pages/CartPage.py
from playwright.sync_api import Page


class CartPage:
    PRODUCTS_PAGE_LINK = 'a[href="/products"]'
    CART_PAGE_LINK = 'a[href="/view_cart"]'
    PROCEED_TO_CHECKOUT = 'a:has-text("Proceed To Checkout")'
    MODAL_MESSAGE = 'h4.modal-title.w-100'
    MODAL_CLOSE_BUTTON = 'button[data-dismiss="modal"]'
    CHECKOUT_PAGE_HEADER = 'h2:has-text("Address Details")'

    def __init__(self, page: Page):
        self.page = page

    def add_product_to_cart(self, product_id):
        self.page.click(f'a[data-product-id="{product_id}"]')

    def open_products_page(self):
        self.page.click(self.PRODUCTS_PAGE_LINK)

    def go_to_cart(self):
        self.page.click(self.CART_PAGE_LINK)

    def proceed_to_checkout(self):
        self.page.click(self.PROCEED_TO_CHECKOUT)

    def is_product_added(self):
        modal_message = self.page.locator(self.MODAL_MESSAGE).text_content()
        self.page.click(self.MODAL_CLOSE_BUTTON)
        return modal_message == "Added!"

    def is_checkout_page(self):
        return self.page.locator(self.CHECKOUT_PAGE_HEADER).is_visible()

File: cypress_framework/cypress/pages/CartPage.js
class CartPage {
    constructor() {
      this.productsPageLink = 'a[href="/products"]';
      this.cartPageLink = 'a[href="/view_cart"]';
      this.proceedToCheckoutButton = 'a:contains("Proceed To Checkout")';
      this.modalMessage = 'h4.modal-title.w-100';
      this.modalCloseButton = 'button[data-dismiss="modal"]';
      this.checkoutPageHeader = 'h2:contains("Address Details")';
    }
  
    addProductToCart(productId) {
      cy.get(`a[data-product-id="${productId}"]`).first().click();
    }
  
    openProductsPage() {
      cy.get(this.productsPageLink).click();
    }
  
    goToCart() {
      cy.get(this.cartPageLink).first().click();
    }
  
    proceedToCheckout() {
      cy.get(this.proceedToCheckoutButton).click();
    }
  
    isProductAdded() {
      cy.get(this.modalMessage)
        .invoke('text')
        .then((text) => {
          expect(text.trim()).to.eq("Added!");
          cy.get(this.modalCloseButton).click();
        });
    }
  
    isCheckoutPage() {
      cy.get(this.checkoutPageHeader).should('be.visible');
    }
  }
  
  export default CartPage;
  